src/components/PostDetails/PostDetails.tsx

```diff
@@ -0,0 +1,142 @@
+import { deleteComment } from '../../api/posts';
+import { Comment } from '../../types/Comment';
+import { Post } from '../../types/Post';
+import { Loader } from '../Loader';
+import { NewCommentForm } from '../NewCommentForm/NewCommentForm';
+
+type Props = {
+  comments: Comment[],
+  selectedPost: Post | null,
+  isErrorComments: boolean,
+  isLoadingComments: boolean,
+  isFormShown: boolean,
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>,
+  setIsErrorComments: (v: boolean) => void,
+  setIsFormShown: (v: boolean) => void,
+};
+
+export const PostDetails: React.FC<Props> = ({
+  comments,
+  selectedPost,
+  isErrorComments,
+  isLoadingComments,
+  isFormShown,
+  setComments,
+  setIsErrorComments,
+  setIsFormShown,
+}) => {
+  const handleDeleteComment = (commentId: number) => {
+    setComments(currentComments => currentComments.filter(
```

Comment:
in case an error happen you anyway change the local state
do it in `then` callback as you do in other places